Find a functionally equivalent code:android.net.LocalSocketImpl.SocketInputStream.available:COMMENT
Method Modifier: public      
Comment:/**
 * {@inheritDoc}
 */

@@ -1,3 +1,6 @@
 {
-    return available_native(fd);
+    FileDescriptor myFd = fd;
+    if (myFd == null)
+        throw new IOException("socket closed");
+    return available_native(myFd);
 }

